### PR TITLE
Refactor: Make SequentialFile and RandomAccessFile as derived classes of InputStream

### LINKED ISSUE #4241: Make RandomAccess a derived class of io::SeekableInputStream
Make [RandomAccessFile](https://github.com/StarRocks/starrocks/blob/fa1d0bf1e879a0ec3624364918197ce1ec58c247/be/src/env/env.h#L233) a derived class of [io::SeekableInputStream](https://github.com/StarRocks/starrocks/blob/fa1d0bf1e879a0ec3624364918197ce1ec58c247/be/src/io/seekable_input_stream.h#L9) can remove some duplicate code and make the IO system more clear.

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

## Which issues of this PR fixes ：

Fixes #4241 #4242 

## Problem Summary(Required) ：

Make SequentialFile and RandomAccessFile as derived classes of InputStream. This would remove many duplicated codes and make the IO stack clearer.

### CODE REVIEW

**Comment by sduzh** [comment]
run starrocks_clang-format

**Comment by sduzh** [comment]
PTAL @chaoyli @imay @wyb

**Review by imay — DISMISSED** [review]

File: be/src/env/env_stream_pipe.cpp [hunk 1]
```diff
@@ -25,8 +25,9 @@ Status StreamPipeSequentialFile::read_one_message(std::unique_ptr<uint8_t[]>* bu
     return _file->read_one_message(buf, length, padding);
 }
 
-Status StreamPipeSequentialFile::skip(uint64_t n) {
-    return _file->seek(n);
+Status StreamPipeSequentialFile::skip(int64_t n) {
+    std::unique_ptr<char[]> buf(new char[n]);
+    return read(buf.get(), n).status();
```
Comment (wyb) [inline be/src/env/env_stream_pipe.cpp:30]: Need to ensure that `n` is actually read.

File: be/src/env/env_stream_pipe.h [hunk 2]
```diff
@@ -10,7 +10,8 @@ class StreamLoadPipe;
 // Since the StreamPipe does not have the standard
 // SequentialFile interface. The StreamPipeSequentialFile
 // is used to wrap the memory to provide the sequential API.
-class StreamPipeSequentialFile : public SequentialFile {
```
Comment (wyb) [inline be/src/env/env_stream_pipe.h:13]: StreamPipeSequentialFile -> PipeInputStream?

File: be/src/env/env_stream_pipe.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/env/env_stream_pipe.cpp:30]: Good finding!

File: be/src/env/env_stream_pipe.h (hunk 2 above)
Comment (sduzh) [reply be/src/env/env_stream_pipe.h:13]: Good suggestion! How do you think about `StreaLoadPipeInputStream`?

**Review by wyb — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**